mesheryctl/internal/cli/root/system/context.go

```diff
@@ -0,0 +1,399 @@
+		if err != nil {
+			return err
+		}
+		_, exists := configuration.Contexts[args[0]]
+		if !exists {
+			return errors.New("no context to delete")
+		}
+
+		if viper.GetString("current-context") == args[0] {
+			var res bool
+			if utils.SilentFlag {
+				res = true
+			} else {
+				res = utils.AskForConfirmation("Are you sure you want to delete the current context")
+			}
+
+			if !res {
+				log.Printf("Delete aborted")
+				return nil
+			}
+
+			var result string
+
+			if newContext != "" {
+				_, exists := configuration.Contexts[newContext]
+				if !exists {
+					return errors.New("new context wrongly set")
+				}
+
+				if newContext == args[0] {
+					return errors.New("choose a new context other than the context being deleted")
+				}
+
+				result = newContext
+			} else {
+				var listContexts []string
+				for context := range configuration.Contexts {
+					if context != args[0] {
+						listContexts = append(listContexts, context)
+					}
+				}
+
+				prompt := promptui.Select{
+					Label: "Select context",
+					Items: listContexts,
+				}
+
+				_, result, err = prompt.Run()
+
+				if err != nil {
+					fmt.Printf("Prompt failed %v\n", err)
+					return err
+				}
+			}
+
+			fmt.Printf("The current context is now %q\n", result)
+			viper.Set("current-context", result)
+		}
+		delete(configuration.Contexts, args[0])
+		viper.Set("contexts", configuration.Contexts)
+		log.Printf("deleted context %s", args[0])
+		err = viper.WriteConfig()
+
+		return err
+	},
+}
+
+// listContextCmd represents the list command
+var listContextCmd = &cobra.Command{
+	Use:          "list",
+	Short:        "list contexts",
+	Long:         `List current context and available contexts`,
+	Args:         cobra.NoArgs,
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := viper.Unmarshal(&configuration)
+		if err != nil {
+			return err
+		}
+		var contexts = configuration.Contexts
+		if contexts == nil {
+			// return errors.New("no available contexts")
+			log.Print("No contexts available. Use `mesheryctl system context create <name>` to create a new Meshery deployment context.\n")
+			return nil
+		}
+
+		if currContext == "" {
+			currContext = viper.GetString("current-context")
+		}
+		if currContext == "" {
+			log.Print("Current context not set\n")
+		} else {
+			log.Printf("Current context: %s\n", currContext)
+		}
+		log.Print("Available contexts:\n")
+
+		//sorting the contexts to get a consistent order on each subsequent run
+		keys := make([]string, 0, len(contexts))
+		for k := range contexts {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+
+		for _, k := range keys {
+			log.Printf("- %s", k)
+		}
+
+		if currContext == "" {
+			log.Print("\nRun `mesheryctl system context switch <context name>` to set the current context.")
+		}
+		return nil
+	},
+}
+
+// viewContextCmd represents the view command
+var viewContextCmd = &cobra.Command{
+	Use:          "view",
```

Comment:
here too!

Reply:
@piyushsingariya  I think you are suggesting to  add examples here? `Use` is okay? This is what we doing
in other commands

Reply:
we recently are trying to improve the ux for mesheryctl, and improving `Use` in the commands.